Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -647,546 +552,351 @@ protected virtual void Dispose(bool disposing)
+        public bool HasRemoteAccess(IPAddress remoteIp)
         {
-            var str = address.ToString();
-            if (address.AddressFamily == AddressFamily.InterNetworkV6)
+            var config = _configurationManager.GetNetworkConfiguration();
+            if (config.EnableRemoteAccess)
             {
-                int i = str.IndexOf("%", StringComparison.OrdinalIgnoreCase);
-                if (i != -1)
+                // Comma separated list of IP addresses or IP/netmask entries for networks that will be allowed to connect remotely.
+                // If left blank, all remote addresses will be allowed.
+                if (_remoteAddressFilter.Any() && !_lanSubnets.Any(x => x.Contains(remoteIp)))
                 {
-                    str = str.Substring(0, i);
-                }
+                    // remoteAddressFilter is a whitelist or blacklist.
+                    var matches = _remoteAddressFilter.Count(remoteNetwork => remoteNetwork.Contains(remoteIp));
+                    if ((!config.IsRemoteIPFilterBlacklist && matches > 0)
+                        || (config.IsRemoteIPFilterBlacklist && matches == 0))
+                    {
+                        return true;
+                    }
 
-                return $"[{str}]";
+                    return false;
+                }
+            }
+            else if (!_lanSubnets.Any(x => x.Contains(remoteIp)))
+            {
+                // Remote not enabled. So everyone should be LAN.
+                return false;
             }
 
-            return str;
+            return true;
         }
 
-        private void ConfigurationUpdated(object? sender, ConfigurationUpdateEventArgs evt)
+        /// <inheritdoc/>
+        public IReadOnlyList<PhysicalAddress> GetMacAddresses()
         {
-            if (evt.Key.Equals(NetworkConfigurationStore.StoreKey, StringComparison.Ordinal))
-            {
-                UpdateSettings((NetworkConfiguration)evt.NewConfiguration);
-            }
+            // Populated in construction - so always has values.
+            return _macAddresses;
         }
 
-        /// <summary>
-        /// Checks the string to see if it matches any interface names.
-        /// </summary>
-        /// <param name="token">String to check.</param>
-        /// <param name="index">Interface index numbers that match.</param>
-        /// <returns><c>true</c> if an interface name matches the token, <c>False</c> otherwise.</returns>
-        private bool TryGetInterfaces(string token, [NotNullWhen(true)] out List<int>? index)
+        /// <inheritdoc/>
+        public IReadOnlyList<IPData> GetLoopbacks()
         {
-            index = null;
-
-            // Is it the name of an interface (windows) eg, Wireless LAN adapter Wireless Network Connection 1.
-            // Null check required here for automated testing.
-            if (_interfaceNames is not null && token.Length > 1)
+            var loopbackNetworks = new List<IPData>();
```

Comment:
You could create a "fast path" and avoid an allocation if you check that both are disabled and return `Array.Empty`